src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -161,6 +163,33 @@ CHIP_ERROR CodegenDataModelProvider::Startup(DataModel::InteractionModelContext
 
     InitDataModelForTesting();
 
+    // If no ConfigurationVersion was ever set, initialize it to 1.
+    chip::StorageKeyName kStorageKey = chip::DefaultStorageKeyAllocator::ConfigurationVersion();
```

Comment:
This is not really going to work very well for things like Matter.framework, where there is no global key storage.  In particular, all the writes here will fail (which works out because this code ignores write failures) and the reads will always claim the key is not found.

In general, we should try pretty hard to not hardcode global storage keys.  Instead we should have delegate APIs, possibly with a default implementation that uses a global storage key but can be overridden to do something else.  That's how things like session resumption storage and whatnot are structured...

And anyway, we don't want every single data model provider (and we _are_ working on having other ones) have code like this.... It should live somewhere else.

Reply:
> In general, we should try pretty hard to not hardcode global storage keys. Instead we should have delegate APIs, possibly with a default implementation that uses a global storage key but can be overridden to do something else. That's how things like session resumption storage and whatnot are structured...

I need to look into this, but sounds like a reasonable addition.